spec/factories/strategies/valkyrie_resource.rb

```diff
@@ -9,3 +12,20 @@ def result(evaluation)
     result
   end
 end
+
+# This strategy is registered in spec_helper.rb and given a name that can be called by factory bot (e.g. create_using_test_adapter).
+# The test adapter is registered in spec_helper.rb (e.g. in-memory adapter).
+# See spec_helper.rb to confirm registered name of the strategy and the test adapter registration.
+# Example:  let(:resource) { FactoryBot.create_using_test_adapter(:hyrax_work) }
+class ValkyrieTestAdapterCreateStrategy
```

Comment:
can we refactor to deduplicate?

Reply:
I looked at a way to pass the adapter in and couldn't find one.  Without that ability, then we have to have both strategies.  If you know of a way to pass in the adapter, I am happy to refactor this from two strategies into one.

Reply:
I'm happy to have both strategies, just not to have them be almost exclusively duplicate code.

Couldn't we solve this with sub-classing and a `#persister` method?

Reply:
PENDING response

Reply:
DONE - subclassed